site/src/test/java/uk/nhs/digital/ps/beans/navigation/CiBreadcrumbTest.java

```diff
@@ -0,0 +1,236 @@
+        given(ciLandingBean.getTitle()).willReturn(TITLE_CI_LANDING);
+    }
+
+    private void mockPublicationHasParentSeries(Publication publication) {
+        given(publication.getParentDocument()).willReturn(series);
+        given(series.getTitle()).willReturn(TITLE_SERIES);
+    }
+
+    private void mockDatasetHasParentPublication(Dataset dataset) {
+        given(dataset.getParentPublication()).willReturn(publication);
+        given(publication.getTitle()).willReturn(TITLE_PUBLICATION);
+    }
+
+    private CiBreadcrumb getCiBreadcrumb() {
+        CiBreadcrumbProvider ciBreadcrumbProvider = new CiBreadcrumbProvider(hstRequest);
+        return ciBreadcrumbProvider.getBreadcrumb();
+    }
+
+    private void assertBreadcrumbsMatch(List<String> expected, CiBreadcrumb ciBreadcrumb) {
+
+        List<String> actual = ciBreadcrumb.getItems().stream()
+            .map(BreadcrumbItem::getTitle)
+            .collect(Collectors.toList());
+
+        assertEquals("Breadcrumbs match", expected, actual);
+    }
+
+    @Test
+    public void displaysBreadcrumb_CiLanding() throws Exception {
+
+        // given
+        mockCiLanding();
+        mockCurrentDocumentBean(CiLanding.class, TITLE_CI_LANDING);
+        List<String> expected = Collections.singletonList(TITLE_CI_LANDING);
+
+        // when
+        CiBreadcrumb ciBreadcrumb = getCiBreadcrumb();
+
+        // then
+        assertBreadcrumbsMatch(expected, ciBreadcrumb);
+    }
+
+    @Test
+    public void displaysBreadcrumb_Series() throws Exception {
+
+        // given
+        mockCurrentDocumentBean(Series.class, TITLE_SERIES);
+        List<String> expected = Collections.singletonList(TITLE_SERIES);
+
+        // when
+        CiBreadcrumb ciBreadcrumb = getCiBreadcrumb();
+
+        // then
+        assertBreadcrumbsMatch(expected, ciBreadcrumb);
+    }
+
+    @Test
+    public void displaysBreadcrumb_Archive() throws Exception {
+
+        // given
+        mockCurrentDocumentBean(Archive.class, TITLE_ARCHIVE);
+        List<String> expected = Collections.singletonList(TITLE_ARCHIVE);
+
+        // when
+        CiBreadcrumb ciBreadcrumb = getCiBreadcrumb();
+
+        // then
+        assertBreadcrumbsMatch(expected, ciBreadcrumb);
+    }
+
+    @Test
+    public void displaysBreadcrumb_Publication() throws Exception {
+
+        // given
+        mockCurrentDocumentBean(Publication.class, TITLE_PUBLICATION);
+        List<String> expected = Collections.singletonList(TITLE_PUBLICATION);
+
+        // when
+        CiBreadcrumb ciBreadcrumb = getCiBreadcrumb();
+
+        // then
+        assertBreadcrumbsMatch(expected, ciBreadcrumb);
+    }
+
+    @Test
+    public void displaysBreadcrumb_Series_Publication() throws Exception {
+
+        // given
+        mockCurrentDocumentBean(Publication.class, TITLE_PUBLICATION);
+        mockPublicationHasParentSeries((Publication) currentDocumentBean);
```

Comment:
It might be better to make mockCurrentDocumentBean return the mocked object and not have the class field at all. Then the mockCurrentDocumentBean method could return the correct class of document using generics